### PR TITLE
finished day 6 before CR

### PR DESCRIPTION
**Pull request by idnv**

### CODE REVIEW

File: Day6/day6-part2.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+input.split(crossPlatformEndOfFile).forEach(row => treeMap[row.split(')')[1]] = row.split(')')[0]);
+// set source and dest
+const source = "YOU";
+const destination = "SAN";
+const root = getRootValue(treeMap);
+// calc paths from source and dest to root 
+const pathFromSource = calcShortPathToRoot(treeMap, root, source);
+const pathFromDest = calcShortPathToRoot(treeMap, root, destination);
+// find first common father value
+const commonFather = pathFromSource.filter(element => pathFromDest.includes(element)).shift();
```
Comment (orvi) [inline Day6/day6-part2.js]: You can use `find` to get the first element for a prediction

File: Day6/day6-part2.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// calculate answer
+var answer = pathFromSource.length + pathFromDest.length;
+
+
+function getRootValue(tree) {
+    const isChildren = (p) => treeMap.hasOwnProperty(p);
+    return Object.entries(tree).map(([_, parent]) => parent).find(p => !isChildren(p));
+}
+
+function calcShortPathToRoot(tree, root, node, path = []) {
```
Comment (orvi) [inline Day6/day6-part2.js]: the `path` parameter is not needed. Use the returning value of the function to keep track of the path

File: Day6/day6-part2.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+// find first common father value
+const commonFather = pathFromSource.filter(element => pathFromDest.includes(element)).shift();
+//short path to common father from source
+const indexOfCommonFatherInSourcePath = pathFromSource.indexOf(commonFather);
+pathFromSource.splice(indexOfCommonFatherInSourcePath, pathFromSource.length - indexOfCommonFatherInSourcePath);
+//short path to common father from destination
+const indexOfCommonFatherInDestPath = pathFromDest.indexOf(commonFather);
+pathFromDest.splice(indexOfCommonFatherInDestPath, pathFromDest.length - indexOfCommonFatherInDestPath);
+// calculate answer
+var answer = pathFromSource.length + pathFromDest.length;
```
Comment (orvi) [inline Day6/day6-part2.js]: `splice` might be an expensive operation. Can you calculate the answer without using it?

**Review by orvi — CHANGES_REQUESTED** [review]

File: Day6/day6-part2.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    .readFileSync(calcedPath)
+    .toString()
+
+//take input and make it a dictinioary [node] = fatherNode
+var treeMap = new Object();
+const crossPlatformEndOfFile = require('os').EOL;
+input.split(crossPlatformEndOfFile).forEach(row => treeMap[row.split(')')[1]] = row.split(')')[0]);
+const root = getRootValue(treeMap);
+// calc paths from source and dest to root 
+const pathFromSource = calcShortPathToRoot(treeMap, root, treeMap[source]);
```
Comment (orvi) [inline Day6/day6-part2.js:20]: does it work if source==root?

File: Day6/day6-part2.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+// set source and dest
+const source = "COM";
+const destination = "COM";
+
+// claculate cross platform path
+const path = require('path');
+const calcedPath = path.join('Day6', 'input2.txt');
```
Comment (orvi) [inline Day6/day6-part2.js]: can't find the file "input2"

Comment (idnv) [reply Day6/day6-part2.js]: ohh

**Review by orvi — APPROVED** [review]

**Pull request merged**